### ISSUE TITLE
[Feature Request] option for mouse trapping while pressing WIN key

### ISSUE THREAD

**Issue by beppe9000** [cairoshell/cairoshell#258]
When using maximized windows with custom UI and tiny borders is often difficult to interact with it and instead i misclick the top cairo bar.

So I was thinking, 'why not use the windows key to fix that problem?'. 

What I propose here is to snap the mouse cursor to the client area of the desktop (as if the screen ended there) when the WIN key (or better a configurable option set by default to windows key) is held down (maybe with some other optional modifier).

In other words to lock the Y coordinate of the mouse cursor to Y>[cairo top bar height].

**Comment by dremin** [comment]
This should be implemented by a window manager outside of the scope of Cairo.
